evap/staff/views.py

```diff
@@ -681,27 +684,39 @@ def semester_import(request, semester_id):
 
 
 @manager_required
-def semester_export(request, semester_id):
-    semester = get_object_or_404(Semester, id=semester_id)
+class SemesterExportView(SingleObjectMixin, FormsetView):
+    model = Semester
+    pk_url_kwarg = "semester_id"
+    formset_class = formset_factory(form=ExportSheetForm, can_delete=True, extra=0, min_num=1, validate_min=True)
+    template_name = "staff_semester_export.html"
 
-    ExportSheetFormset = formset_factory(form=ExportSheetForm, can_delete=True, extra=0, min_num=1, validate_min=True)
-    formset = ExportSheetFormset(request.POST or None, form_kwargs={"semester": semester})
+    object: Semester
```

Comment:
We talked about this and got confused. Maybe it's better to keep the old view here?